src/main/java/ca/warp7/frc2022/subsystems/DriveTrain.java

```diff
@@ -39,7 +41,12 @@ public static void setVariant(DriveTrainVariant variant) {
         if (driveTrainVariant != null) {
             throw new IllegalStateException("Cannot set drive train variant");
         }
-        driveTrainVariant = variant;
+        if (kEnableDriveTrain){
+            driveTrainVariant = variant;
+        }
+        else{
+            driveTrainVariant = new LazyDriveTrainVariant();
+        }
     }
 
     private final LazySolenoid shifterSolenoid =
```

Comment:
The LazySolenoid here also needs a be tied into the kEnableDriveTrain

Also, are we missing something like a kEnableDriveTrainShifter constant?  i.e., should this be

`new LazySolenoid(kDriveShifterID, kEnableSolenoids && kEnableDriveTrain && kEnableDriveTrainShifter);`